### PR TITLE
New chr_length reference converter

### PR DESCRIPTION
**Pull request by jasonwalker80**
Add chr_length converter algorithm to match chromosome names when all chromosome lengths are unique and match between reference builds.

### CODE REVIEW

**Comment by jasonwalker80** [comment]
I can already think of one issue.  What if there are multiple chromosomes with the same length? Suggestions welcome.

File: lib/perl/Genome/Model/Build/ReferenceSequence/Converter.pm [hunk 1]
```diff
@@ -40,6 +40,10 @@ class Genome::Model::Build::ReferenceSequence::Converter {
             is_optional => 1,
         },
     ],
+    has_transient_optional => [
+        _chr_map_hash_ref => {
+        },
```
Comment (tmooney) [inline lib/perl/Genome/Model/Build/ReferenceSequence/Converter.pm]: It'd be nice if this property included:
```perl
is => 'HASH'
```
(The type needn't necessarily be included in the name then, but either way!)

A `doc` line might be good, too, but is less important for internal things.

**Comment by tmooney** [comment]
The `seqdict.sam` does include the MD5s, so we could use those to help properly match.  I don't think there's already a method to pull them out of the file, though.

**Comment by jasonwalker80** [comment]
I'm already on the md5 solution.

**Pull request closed without merging**